### DISCUSSION TITLE
[Feature] Android App - specifiy SSIDs for automatic syncing

### DISCUSSION THREAD

**Discussion by MPatzek** [immich-app/immich#4176] [Feature Request]
### The feature

If the automatic backup is only allowed in WLAN connection, it would be good to allow a spcification of SSIDs in which the images are backuped. This is especially relevant if the immich server is only reachable in LAN.

### Platform

- [ ] Server
- [ ] Web
- [X] Mobile

**Comment by whc2001** [comment]
I second this and #2335 (although is one is less on priority for me since I have DNS rewrite). Here is what we have in HomeAssistant, which automatically detects home SSID, that unlocks the app and uses the internal URL for connection. Considering the backup process of photos and videos will consume quite a lot of data traffic it would make sense to prefer backing up inside the same LAN as the server hosted.

![Screenshot_20231201-233422_Home Assistant](https://github.com/immich-app/immich/assets/16266909/17e2769b-8ea2-4d84-9cb6-22c66c042567)

**Reply by popy2k14** [reply]
+1 for this

**Comment by ryanwinter** [comment]
I would like to see a breakdown by image/video, or potentially file size. The scenario I have is that the endpoint is available everywhere, but cloudflare doesnt allow files larger than 100MB to sync, so it fails. I would still like smaller files to upload no matter what SSID its connected too, however I would like to limit video uploads to only one of my home SSIDs.

**Comment by groot-stuff** [comment]
+1 for this.  
I have it set to WiFi + Charging only which limits syncing to when I'm home most of the time, but syncs still fail if I'm on WiFi elsewhere and plug into a portable charger.
My Immich instance (all services actually) are LAN-only, my limited expertise and fear of the WWW keep me from exposing anything on the WAN side.
